sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -207,6 +207,16 @@ class SessionCatalog(
           "you cannot create a database with this name.")
     }
     validateName(dbName)
+    // SPARK-25464 fail if DB location exists and is not empty
+    val dbPath = new Path(dbDefinition.locationUri)
+    val fs = dbPath.getFileSystem(hadoopConf)
+    if (fs.exists(dbPath)) {
+      val fileStatus = fs.listStatus(dbPath)
+      if (0 != fileStatus.length) {
```

Comment:
`if (fileStatus.nonEmpty)`? at least, flip the comparison. You can add the location to the exception message, maybe.